src/main/groovy/io/github/cosmicsilence/scalafix/ScalafixExtension.groovy

```diff
@@ -38,6 +40,16 @@ class ScalafixExtension {
     ScalafixExtension(Project project) {
         this.project = project
         configFile = project.objects.fileProperty()
+
+        def projectConfigFile = project.getLayout().getProjectDirectory().file(DEFAULT_CONFIG_FILE)
+        def rootProjectConfigFile = project.rootProject.getLayout().getProjectDirectory().file(DEFAULT_CONFIG_FILE)
+
+        if (projectConfigFile.asFile.exists()) {
+            configFile = project.objects.fileProperty().convention(projectConfigFile)
+        } else if (rootProjectConfigFile.asFile.exists()) {
+            configFile = project.objects.fileProperty().convention(rootProjectConfigFile)
+        }
+
```

Comment:
how about something like this:

```diff
     ScalafixExtension(Project project) {
         this.project = project
-        configFile = project.objects.fileProperty()
+        configFile = project.objects.fileProperty().convention(
+            locateConfigFile(project) ?: locateConfigFile(project.rootProject)
+        )
         includes = project.objects.setProperty(String)
         excludes = project.objects.setProperty(String)
     }

+    private RegularFile locateConfigFile(Project project) {
+        def configFile = project.getLayout().getProjectDirectory().file(DEFAULT_CONFIG_FILE)
+        return (configFile.asFile.exists() && configFile.asFile.isFile()) ? configFile : null
+    }
+
```
I find it easier to read and avoids some duplication